src/main/java/fi/tkgwf/ruuvi/common/parser/impl/DataFormat5Parser.java

```diff
@@ -54,7 +54,7 @@ public RuuviMeasurement parse(byte[] data) {
         if (!ByteUtils.isMaxUnsignedByte(data[15])) {
             m.setMovementCounter(data[15] & 0xFF);
         }
-        if (!ByteUtils.isMaxSignedShort(data[16], data[17])) {
+        if (!ByteUtils.isMaxUnsignedShort(data[16], data[17])) {
```

Comment:
I think you have an honest typo here -- measurement sequence number should be unsigned:

> 

Measurement sequence number (16bit unsigned), each time a measurement is taken, this is incremented by one, used for measurement de-duplication (depending on the transmit interval, multiple packets with the same measurements can be sent, and there may be measurements that never were sent)

From the [specs](https://github.com/ruuvi/ruuvi-sensor-protocols):

> Measurement sequence number (16bit unsigned), each time a measurement is taken, this is incremented by one, used for measurement de-duplication (depending on the transmit interval, multiple packets with the same measurements can be sent, and there may be measurements that never were sent)

And..

> Not available is signified by largest presentable number for unsigned values, smallest presentable number for signed values and all bits set for mac.

and in the table of examples:

> 65535 | Invalid / not available

Reply:
Yes, it definitely should be max unsigned